src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,480 @@
+    slug: z.string(),
+    image: z.string().optional(),
+    tags: z.array(z.string()),
+  },
+});
+
+export const collections = { blog };
+```
+
+`await getCollection('blog')` will return entries of the following type, with the type of `data` inferred from your schema:
+
+```ts
+{
+  // Unique ID using file path relative to src/content/[collection]
+  // Ex. for content/[collection]/file-1.md...
+  id: 'file-1.md' | 'file-2.md' | ...;
+	// URL-ready slug using ID with file extension stripped
+  // Ex. for content/[collection]/file-1.md...
+	slug: 'file-1' | 'file-2' | ...;
+  // Inferred from collection schema
+  // Defaults to `any` if no schema is configured
+  data: {
+    title: string;
+    slug: string;
+    image?: string;
+    tags: string[];
+  };
+  // Raw body of the Markdown or MDX document
+  body: string;
+}
+```
+
+:::note
+The `body` is the *raw* content of the file. This ensures builds remain performant by avoiding expensive rendering pipelines. See [**Rendering content**](#rendering-content-with-renderentry) to compile this body to a `Content` component for use in your Astro files.
+:::
+
+### Getting from nested directories
+
+[As noted earlier](#organizing-with-nested-directories), you may organize entries into directories as well. The result will **still be a flat array** when getting a collection via `getCollection`, with the nested directory reflected in an entry’s `id`:
+
+```ts
+const docsEntries = await getCollection('docs');
+console.log(docsEntries)
+/*
+[
+	{ id: 'en/getting-started.md', slug: 'en/getting-started', data: {...} },
+	{ id: 'en/structure.md', slug: 'en/structure', data: {...} },
+	{ id: 'es/getting-started.md', slug: 'es/getting-started', data: {...} },
+	{ id: 'es/structure.md', slug: 'es/structure', data: {...} },
+	...
+]
+*/
+```
+
+To fetch from a specific directory **within** a collection, Astro recommends `getCollection` with a filter:
+
+```astro
+---
+import { getCollection } from 'astro:content';
+
+const enDocs = await getCollection('docs', ({ id }) => {
+  // Where `id` is 'en/page-1.md' | 'en/page-2.md' | ...
+  return id.startsWith('en');
+});
+---
+```
+
+## Rendering content with `renderEntry`
+
+You may need to render the contents of these Markdown and MDX entries as well. This is especially useful when generating live URLs from your content entries (see [Generating pages from content collections](#generating-pages-from-content-collections)). 
+
+You can retrieve a `Content` component for use in your Astro files with `renderEntry`:
+
+```astro "renderEntry"
+---
+import { renderEntry, getEntry } from 'astro:content';
+
+const announcementPost = await getEntry('announcements', 'welcome.md');
+// Pass a `getEntry` result or `getCollection` array item:
+const { Content } = await renderEntry(announcementPost);
+// Or pass a valid ID and collection name individually:
+const { Content } = await renderEntry({
+  id: announcementPost.id,
+  collection: announcementPost.collection,
+});
```

Comment:
There is! It's more clear when using `getStaticPaths`, where you may destructure individual properties for use as props instead of passing down the entire entry. I agree this doesn't show the need very clearly though. I vote axing this and leaving it to the API reference 👍